packages/lesswrong/components/form-components/SocialPreviewUpload.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import {Components, getSiteUrl, registerComponent, sanitize } from '../../lib/vulcan-lib';
+import { siteImageSetting } from '../vulcan-core/App';
+import { htmlToText } from 'html-to-text'
+import { truncate } from '../../lib/editor/ellipsize';
+import { getPostDescription } from '../posts/PostsPage/PostsPage';
+import { PLAINTEXT_DESCRIPTION_LENGTH, PLAINTEXT_HTML_TRUNCATION_LENGTH } from '../../lib/collections/revisions/collection';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+  },
+  preview: {
+    padding: 16,
+    marginLeft: 10,
+    backgroundColor: theme.palette.grey[100],
+    borderRadius: 6,
+    width: 'min-content',
+  },
+  title: {
+    fontSize: 14,
+    minHeight: 18,
+    fontWeight: 700,
+    marginTop: 12,
+    marginBottom: 7,
+  },
+  description: {
+    fontSize: 12,
+    minHeight: 36,
+    marginBottom: 7,
+  },
+  url: {
+    fontSize: 12,
+    color: theme.palette.grey[600],
+  },
+  blurb: {
+    paddingLeft: 16,
+    paddingRight: 16,
+    fontSize: 14,
+  }
+});
+
+// Derived from PostsEditQueryFragment, but with extra data that can be set locally (only dataWithDiscardedSuggestions)
+interface PostsEditWithLocalData extends PostsEdit { // fragment on Posts
+  readonly contents: RevisionEdit & { dataWithDiscardedSuggestions: string | null} | null,
+}
+
+/**
+ * Build the preview description and extract the first image in the document to use as a fallback. This is duplicating
+ * the fairly roundabout process that happens on the server to generate the preview description and image. The logic here
+ * is _almost_ exactly the same, but skips some overly complicated or potentially slow steps, namely:
+ * - trimming whitespace
+ * - rendering LaTex
+ * - handling markdown and DraftJS (seeing as these are no longer supported for non admins)
```

Comment:
FYI all users can still select markdown, via an account setting

Reply:
Thanks, made it work with markdown now 👍

Reply:
Nice thanks! Is this comment now out-of-date?